polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx

```diff
@@ -137,11 +138,17 @@ export class PositionedOverlay extends PureComponent<
       zIndexOverride,
     } = this.props;
 
+    const nextTop = top == null || isNaN(top) ? undefined : top;
+    const nextLeft = left == null || isNaN(left) ? undefined : left;
+    const nextRight = right == null || isNaN(right) ? undefined : right;
+    const nextWidth = width == null || isNaN(width) ? undefined : width;
+
     const style = {
-      top: top == null || isNaN(top) ? undefined : top,
-      left: left == null || isNaN(left) ? undefined : left,
-      right: right == null || isNaN(right) ? undefined : right,
-      width: width == null || isNaN(width) ? undefined : width,
+      '--pc-positioned-overlay-top': nextTop,
```

Comment:
Though this is updated in the class, these properties still need to be set on the style property to avoid cascade issues.